dnf-behave-tests/dnf/dnf-automatic/reboot.feature

```diff
@@ -0,0 +1,101 @@
+@no_installroot
+Feature: dnf-automatic reboots
+
+
+Background:
+Given I delete file "/etc/yum.repos.d/*.repo" with globs
+  And I create file "/etc/dnf/dnf.conf" with
+    """
+    [main]
+    plugins=0
+    """
+
+
+Scenario: dnf-automatic does not reboot when reboot = never
+  Given I create file "/etc/dnf/automatic.conf" with
+    """
+    [commands]
+    reboot = never
+    reboot_command = "echo 'I LOVE REBOOTING'"
```

Comment:
This is a nice way to test the various scenarios. If you also want to test the default command, either cancel the scheduled reboot (`shutdown -c`) or create a temp file with a shell script similar to this echo and bind-mount it over `/usr/sbin/shutdown`.